### PR TITLE
Create tag for optional upstream tracking.

### PR DESCRIPTION
**Pull request by turboaaa**
Check if user adds custom tag, if not create custom tag. Print tag to stdout, then use postconf to update settings.

I chose two headers to check against just in case one is not used for some reason (my general ignorance of emails is showing.)
http://www.postfix.org/header_checks.5.html
https://www.iana.org/assignments/message-headers/message-headers.xhtml

### CODE REVIEW

**Comment by juanluisbaptiste** [comment]
Hi and thanks for the PR, some observations:

I would prefer to rename the new var as SMTP_HEADER_TAG to continue using the naming of the other vars.

If SMTP_HEADER_TAG is not set is better to leave this feature off instead of forcing it with a random value.

Why did you remove and re-add the sasl_passwd file creation code ? please fix the commit so it only adds the new code as described.

**Comment by turboaaa** [comment]
Please excuse my noobish behavior, this is my first time working on github. I accidently added the second sasl_passwd  file creating on the initial PR, and I was trying to correct it. I will see if I can figure out a better method for this.

I will make the other changes you requested.

**Comment by turboaaa** [comment]
Let me know if that works.

File: run.sh [hunk 1]
```diff
@@ -46,11 +46,11 @@ if [ ! -f /etc/postfix/sasl_passwd ]; then
   fi
 fi
 
-#Set header tag
+#Set header tag  
 if [ -z "${SMTP_HEADER_TAG}" ]; then
```
[the remark this answers is not in the document]
Comment (turboaaa) [reply run.sh]: Doh! Of course it does!. Excuse me while I figure out this conversation tool works as well.
```suggestion
if [ ! -z "${SMTP_HEADER_TAG}" ]; then
```

**Comment by juanluisbaptiste** [comment]
Merged.

**Comment by robsonpeixoto** [comment]
A noob question. How this  SMTP_HEADER_TAG can be used?

**Comment by juanluisbaptiste** [comment]
@robsonpeixoto read the README file.

**Pull request merged**